bitwarden_license/test/Commercial.Core.Test/SecretsManager/AccessPolicies/UpdateAccessPolicyCommandTests.cs

```diff
@@ -11,30 +15,290 @@
+    [BitAutoData(AccessPolicyType.ServiceAccountProjectAccessPolicy, PermissionType.RunAsAdmin)]
+    [BitAutoData(AccessPolicyType.ServiceAccountProjectAccessPolicy, PermissionType.RunAsUserWithPermission)]
+    public async Task UpdateAsync_ProjectGrants_PermissionsCheck_Success(
+        AccessPolicyType accessPolicyType,
+        PermissionType permissionType,
+        Guid data,
+        bool read,
+        bool write,
+        Guid userId,
+        Project grantedProject,
+        Group mockGroup,
+        ServiceAccount mockServiceAccount,
+        SutProvider<UpdateAccessPolicyCommand> sutProvider)
+    {
+        sutProvider.GetDependency<ICurrentContext>().AccessSecretsManager(Arg.Any<Guid>()).Returns(true);
+        BaseAccessPolicy policyToReturn = null;
+        switch (accessPolicyType)
+        {
+            case AccessPolicyType.UserProjectAccessPolicy:
+                policyToReturn =
+                    new UserProjectAccessPolicy
+                    {
+                        Id = data,
+                        Read = true,
+                        Write = true,
+                        GrantedProjectId = grantedProject.Id,
+                        GrantedProject = grantedProject,
+                    };
+                break;
+            case AccessPolicyType.GroupProjectAccessPolicy:
+                mockGroup.OrganizationId = grantedProject.OrganizationId;
+                policyToReturn =
+                    new GroupProjectAccessPolicy
+                    {
+                        Id = data,
+                        GrantedProjectId = grantedProject.Id,
+                        Read = true,
+                        Write = true,
+                        Group = mockGroup,
+                        GrantedProject = grantedProject,
+                    };
+                break;
+            case AccessPolicyType.ServiceAccountProjectAccessPolicy:
+                mockServiceAccount.OrganizationId = grantedProject.OrganizationId;
+                policyToReturn = new ServiceAccountProjectAccessPolicy
+                {
+                    Id = data,
+                    GrantedProjectId = grantedProject.Id,
+                    Read = true,
+                    Write = true,
+                    ServiceAccount = mockServiceAccount,
+                    GrantedProject = grantedProject,
+                };
+                break;
+        }
+
+        switch (permissionType)
+        {
+            case PermissionType.RunAsAdmin:
+                sutProvider.GetDependency<ICurrentContext>().OrganizationAdmin(grantedProject.OrganizationId)
+                    .Returns(true);
+                break;
+            case PermissionType.RunAsUserWithPermission:
+                sutProvider.GetDependency<IProjectRepository>().UserHasWriteAccessToProject(grantedProject.Id, userId)
+                    .Returns(true);
+                break;
+        }
```

Comment:
Extract to a function? Re-used in `UpdateAsync_ServiceAccountGrants_PermissionsCheck_Success`. Would also make the test more easily digested.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/79bce819b50d75b5ce021b4b4dfad21dc9c97cd8